### DISCUSSION TITLE
Ways to increase FPS during harvest

### DISCUSSION THREAD

**Discussion by englundjd** [Courseplay/courseplay#7243] [Q&A]
[log.txt](https://github.com/Courseplay/courseplay/files/6603373/log.txt)
I'm a first timer here and have done some googling on this subject but haven't come across any concrete fixes.

I harvest with courseplay running my combines and augerwagons or trailers (in either overloader or combi mode). While courseplay is determining the route to the combine or from combine to the first waypoint of the unload course I go from ~45fps (doesn't seem to matter graphics settings) to sub 10fps, and sometimes this can take a little while(5 to 30 seconds). The issue gets worse as the amount of equipment on those modes increase and it can get to the point of just never running faster then 10fps. Anyone have any ideas on how to alleviate this? Its kind of a game breaking deal for me.

**Comment by Tensuko** [comment]
That is when our pathfinder is running and right now we don't have a way reduce this. (Limited by the Engine).
It should take less time when the path is more easy (mostly when the headland is done 1 or 2 times).

We hope that for the next FS, it will perform better, also we try to have more options to make it perform better.

**Comment by pvaiko** [comment]
> The issue gets worse as the amount of equipment on those modes increase and it can get to the point of just never running faster then 10fps

What amount are we talking about here? How many combi/overloaders?

Turning off fruit avoidance will definitely help, it is just not very realistic.

**Comment by englundjd** [comment]
I usually play on larger maps with 3-4 large combines and 6 to 8 pieces of equipment on combi/overloader.

**Comment by englundjd** [comment]
I turned off fruit avoidance and things seem to be running noticeably better although still jumpy (it's finding the route faster atleast). My issue now though is that the overloader/combi always seems to be stopped in the way of the second combine coming up the course, or if the trailer is full it will just go ahead and try to take a right turn into the combine (if the truck/dump point is to the right of the equipment) and I end up with a collision. That happens quite a bit more often with the fruit avoidance off.

**Reply by pvaiko** [reply]
Would you mind posting a log file here with debug channels 2 and 11 on, with fruit avoidance *on*?

**Reply by Tensuko** [reply]
> with fruit avoidance _on_?

On both, combine and unloader.
It won't help much if the pipe is in the fruit and the unloader have to get into the fruit.

**Reply by pvaiko** [reply]
I still would like to see which part of the pathfinder takes long, the hybrid or the plain A*.

**Reply by englundjd** [reply]
> Would you mind posting a log file here with debug channels 2 and 11 on, with fruit avoidance _on_?

Sorry I'm a noob, how can i go about turning on the debug channels?

**Reply by Tensuko** [reply]
You need the Master Branch (or wait for the next release) then the channels are back working correctly.

**Reply by Tensuko** [reply]
Use 6.4.0.2 from Release page.

**Reply by englundjd** [reply]
[log.txt](https://github.com/Courseplay/courseplay/files/6690224/log.txt)

hopefully this is the info you are after. Let me know if there is anything else you need. Sorry it took a while tonight was the first time in a week I've been home enough to play.

**Comment by MoslessNeo** [comment]
What helped me was leaving a combine with an auger wagon and two haul trucks in each field.
Use four combines in four fields.
It works perfectly, sometimes for up to a day harvesting a 400-hectare field. With avoid plant activated.

The calculation for new routes is less insane than using three combines and two auger wagons in each field.

Certainly for the next generation of FS this will be solved with the best use of the hardware.

**Reply by Tensuko** [reply]
> Certainly for the next generation of FS this will be solved with the best use of the hardware.

We will see...
